apps/server/src/devServer.ts

```diff
@@ -0,0 +1,24 @@
+import react from "@vitejs/plugin-react";
+import { type Express } from "express";
+import path from "node:path";
+
+export const startDevServer = async (app: Express) => {
```

Comment:
can you make this API feels more express middleware like ?


```suggestion
import express, { RequestHandler } from 'express';

/**
 * Install Vite dev server when env is not production
 * This router MUST be installed at the application root, like so:
 *
 *  const app = express();
 *  app.use(widgetsRouter());
 */
export const widgetsRouter = async (): RequestHandler {
    const router = express.Router();
    
    if(process.env.NODE_ENV === "production") {
        return router;
     }

    //....

    router.use(cors());
    router.use("/", vite.middlewares);
    
    return router;
```